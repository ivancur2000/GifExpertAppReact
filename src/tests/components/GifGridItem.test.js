import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
    
    const title = 'Image test';
    const url = 'https://localhost/image1';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe rederizar correctamente el componente', () => {
        expect(wrapper).toMatchSnapshot(); 
    });
    
    test('Debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('debe tener una img igual a la de la URL', () => {
        const img = wrapper.find('img'); 
        expect(img.prop('src')).toBe(url); 
        expect(img.prop('alt')).toBe(title);
    });
    
    test('debe tener la clase animate_fadeIn', () => {
        const div = wrapper.find('div');
        // expect(div.hasClass('animate__fadeIn')).toBeTruthy();
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBeTruthy();

    })
    
});
