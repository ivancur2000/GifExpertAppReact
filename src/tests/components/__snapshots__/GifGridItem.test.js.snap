// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe rederizar correctamente el componente 1`] = `
<div
  className="cardImg animate__animated animate__fadeIn"
>
  <img
    alt="Image test"
    src="https://localhost/image1"
  />
  <p>
    Image test
  </p>
</div>
`;
